### ISSUE TITLE
Is this a false positive in rule ID "200003"?

### ISSUE THREAD

**Issue by shieku** [coreruleset/coreruleset#2908]
Hello.

I use CRS 3.2 in a WAF on Azure Application Gateway.
Some requests violated the following rules:

- RuleId: 200003
- Description: Multipart request body failed strict validation

An example of a request is as follows

- request header

`Content-Type multipart/form-data; boundary =---- WebKitFormBoundary`

- Request Body

```
------ WebKitFormBoundary
Content-Disposition: form-data; name = "contextID"
Content-Type: text/plain

677700898
------ WebKitFormBoundary
Content-Disposition: form-data; name = "CreatedDate"
Content-Type: text/plain

2022 07-06T 13:22:36. 000Z
------ WebKitFormBoundary
Content-Disposition: form-data; name = "FileName"
Content-Type: text/plain

curl.txt
------ WebKitFormBoundary
Content-Disposition: form-data; name = "FileSize"
Content-Type: text/plain

681
------ WebKitFormBoundary
Content-Disposition: form-data; name = "ModifiedDate"
Content-Type: text/plain

2022 07-06T 13:22:37. 000Z
------ WebKitFormBoundary
Content-Disposition: form-data; name = "contents"
Content-Type: text/plain

(omitted for binary data)
------ WebKitFormBoundary - -
```

Is this false positive? I couldn't specify why I violated the rules.
If there is a mistake in the content of the request, please tell me.

Thank you and best regards,

**Comment by franbuehler** [comment]
Hello @shieku!

Thanks for submitting and sorry for your inconvenience.

Unfortunately, you are facing a false positive in the ModSecurity recommended rule [200003](https://github.com/SpiderLabs/ModSecurity/blob/v3/master/modsecurity.conf-recommended#L83) which are not part of the Core Rule Set. 

After taking a quick look at your submitted request body, it might be that the extra spaces are causing some problems: `name = "contents"`. You can remove them and test again.

If it doesn't work, you need to write one or more rule exclusions that are specific to your setup.
If you are not familiar with this technique then take a look at the [tutorial at netnea](https://www.netnea.com/cms/apache-tutorials/) that covers handling false positives.

**Comment by shieku** [comment]
The space in the request body is incorrect. Actually, it is not included.
The correct request is as follows. Are there any other strange places?
- request header
```
Content-Type multipart/form-data; boundary=----WebKitFormBoundary
```

- Request Body
```
------WebKitFormBoundary
Content-Disposition: form-data; name="contextID"
Content-Type: text/plain

677700898
------WebKitFormBoundary
Content-Disposition: form-data; name="CreatedDate"
Content-Type: text/plain

2022-07-06T13:22:36.000Z
------WebKitFormBoundary
Content-Disposition: form-data; name="FileName"
Content-Type: text/plain

curl.txt
------WebKitFormBoundary
Content-Disposition: form-data; name="FileSize"
Content-Type: text/plain

681
------WebKitFormBoundary
Content-Disposition: form-data; name="ModifiedDate"
Content-Type: text/plain

2022-07-06T13:22:37.000Z
------WebKitFormBoundary
Content-Disposition: form-data; name="contents"
Content-Type: text/plain

(omitted for binary data)
------WebKitFormBoundaryO5quBRiT4G7Vm3R7--
```

I will check the information you gave me. Thank you for your help.

**Comment by dune73** [comment]
That looks better, but given this is not a CRS rule and you are probably on the custom implementation on Azure, we think you need to ask their support. 200003 is a complicated rule and not knowing the implementation well enough makes it very hard to help you.

**Comment by dune73** [comment]
Well spotted @airween
